src/Factory/PredisParametersFactory.php

```diff
@@ -24,8 +25,13 @@ public static function create(array $options, string $class, string $dsn): Param
         }
 
         $defaultOptions = ['timeout' => null]; // Allow to be consistent with old version of Predis where default timeout was 5
-        $dsnOptions     = static::parseDsn(new RedisDsn($dsn));
-        $dsnOptions     = array_merge($defaultOptions, $options, $dsnOptions);
+
+        if (isset($options['scan']) && $options['scan'] === 'prefix') {
+            $dsnOptions[Redis::OPT_SCAN] = Redis::SCAN_PREFIX;
```

Comment:
This is overwritten by next instruction

Reply:
You are right, moved after parseDsn() so the options get merged. Resolved at https://github.com/snc/SncRedisBundle/pull/745/commits/95f7e4e6cc6ba8c980358c90dc33a809842bcca2